types/datum.go

```diff
@@ -278,10 +278,12 @@ func (d *Datum) GetMysqlEnum() Enum {
 }
 
 // SetMysqlEnum sets Enum value
-func (d *Datum) SetMysqlEnum(b Enum) {
+func (d *Datum) SetMysqlEnum(b Enum, collation string, length int) {
 	d.k = KindMysqlEnum
 	d.i = int64(b.Value)
 	sink(b.Name)
+	d.collation = collation
+	d.length = uint32(length)
```

Comment:
This may cause bugs, please remove this line. And better to remove it from arguments.